modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -1248,8 +1250,11 @@ public String uploadDocument(String processName, String processVersion, String d
             }
         }
         try {
-            StreamHostObject s = (StreamHostObject) docObject;
-            InputStream docStream = s.getStream();
+            if(docUrl.equals("NA")) {
```

Comment:
Swapping this equals will eliminate NullPointerException when docUrl is null. It's always best to call equals() method on constants.